modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -2596,11 +2596,11 @@ NSFVAction::checkBoundaryParameterErrors()
   {
     if (_inlet_boundaries.size())
     {
-      checkSizeFriendParams(_inlet_boundaries.size(),
+      checkSizeFriendParams(_inlet_boundaries.size() * _passive_scalar_names.size(),
                             _passive_scalar_inlet_types.size(),
```

Comment:
Shouldn't this be a `vector<vector<string>` param too to make sure there is a one-to-one correspondence between this guy and the `_passive_scalar_inlet_function`?

Reply:
we have it as a MultiMooseEnum which has a nicer param list
there's a trade off until we support std::vector<MultiMooseEnum> which would offer both

Reply:
We can still use the MultiMooseEnum to check the input, but in the constructor. However at the moment we have something like this for the inlet functions:

```
passive_scalar_inlet_function = `inlet_1_var_1 inlet_2_var_1; inlet_1_var2 inlet_2_var2`
```

and I think it would look bad with the multimoose enum:

```
passive_scalar_inlet_types = `inlet_1_var_1 inlet_2_var_1 inlet_1_var2 inlet_2_var2`
```
 I think we need to make that a vector of vectors too and loop through all of them.

Also this will not solve the problem which is this:

```
or (unsigned int name_i = 0; name_i < _passive_scalar_names.size(); ++name_i)
  {
    unsigned int flux_bc_counter = 0;
    for (unsigned int bc_ind = 0; bc_ind < _inlet_boundaries.size(); ++bc_ind)
    {
      if (_passive_scalar_inlet_types[bc_ind] == "fixed-value")
      {
```

So we will need:
```
if (_passive_scalar_inlet_types[name_i][bc_ind] == "fixed-value")
      {
``` 

I can push this later today if you are short on time.

Reply:
in the documentation page for the action, if we do not use a multiMooseEnum for the parameter, we will not be telling the user which options are accepted

Reply:
Well, we have been doing this at other instances too:

```
params.addParam<std::vector<std::vector<std::string>>>(
      "friction_types", "The types of friction forces for every block in 'friction_blocks'.");
```

We can still add the options for the parameters manually. Or 
you can do one index [name_i*num_bcs + bc_ind] (or the other way around) but the code I linked needs to be changed either way.

Reply:
well this isnt the best. There s only two options here there though

Reply:
so it makes it easier to specify the types correctly